src/extension_utils/utils.ts

```diff
@@ -372,9 +418,194 @@ export const installPythonDependencies = async (
+        } catch (err) {
+            vscode.window
+                .showErrorMessage(
+                    CONSTANTS.ERROR.NO_PYTHON_PATH,
+                    DialogResponses.INSTALL_PYTHON
+                )
+                .then((selection: vscode.MessageItem | undefined) => {
+                    if (selection === DialogResponses.INSTALL_PYTHON) {
+                        const okAction = () => {
+                            open(CONSTANTS.LINKS.DOWNLOAD_PYTHON);
+                        };
+                        showPrivacyModal(
+                            okAction,
+                            CONSTANTS.INFO.THIRD_PARTY_WEBSITE_PYTHON
+                        );
+                    }
+                });
+            return "";
+        }
+    }
+    // fix path to be absolute
+    if (!path.isAbsolute(originalPythonExecutableName)) {
+        originalPythonExecutableName = path.join(
+            vscode.workspace.rootPath,
+            originalPythonExecutableName
+        );
+    }
+
+    if (!fs.existsSync(originalPythonExecutableName)) {
+        await vscode.window.showErrorMessage(CONSTANTS.ERROR.BAD_PYTHON_PATH);
+        return "";
+    }
+
+    if (!(await validPythonVersion(originalPythonExecutableName))) {
+        return "";
+    }
+
+    pythonExecutableName = originalPythonExecutableName;
+
+    if (!(await checkForDependencies(context, pythonExecutableName))) {
+        // environment needs to install dependencies
+        if (!(await checkIfVenv(context, pythonExecutableName))) {
+            pythonExecutableName = await getPythonVenv(context);
+            if (await hasVenv(context)) {
+                // venv in extention exists with wrong dependencies
+                if (
+                    !(await checkForDependencies(context, pythonExecutableName))
+                ) {
+                    await installDependencies(context, pythonExecutableName);
+                }
+            } else {
+                pythonExecutableName = await promptInstallVenv(
+                    context,
+                    originalPythonExecutableName
+                );
+            }
+        }
+
+        if (pythonExecutableName === originalPythonExecutableName) {
+            // going with original interpreter, either because
+            // already in venv or error in creating custom venv
+            if (checkConfig(CONFIG.SHOW_DEPENDENCY_INSTALL)) {
+                await vscode.window
+                    .showInformationMessage(
+                        CONSTANTS.INFO.INSTALL_PYTHON_DEPS,
+                        DialogResponses.INSTALL_NOW,
+                        DialogResponses.DONT_INSTALL
+                    )
+                    .then(
+                        async (
+                            installChoice: vscode.MessageItem | undefined
+                        ) => {
+                            if (installChoice === DialogResponses.INSTALL_NOW) {
+                                await installDependencies(
```

Comment:
Same as above, do we want to do anything if this fails?

Reply:
same as above also ^